### PR TITLE
add imagerelationship field type

### PR DESCRIPTION
**Pull request by nickhsine**
### Why we need imagerelationship field type?

We can select image by seeing the real image, not just the name or id of the image when we are editing Post or some list related to images.

### CODE REVIEW

**Comment by nickhsine** [comment]
@hcchien please help review this functionality.

File: admin/client/components/ImagesSelection.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+const ImageItem = React.createClass({
+    displayName: 'ImageItem',
+    propTypes: {
+        url: React.PropTypes.string.isRequired,
+        width: React.PropTypes.number.isRequired,
+        padding: React.PropTypes.number,
+        link: React.PropTypes.string,
+        isSelected: React.PropTypes.bool,
+        onClick: React.PropTypes.func
+    },
```
Comment (hcchien) [inline admin/client/components/ImagesSelection.js]: key sorting?

File: admin/client/components/ImagesSelection.js [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+const ImageGrid = React.createClass({
+    displayName: 'ImageGrid',
+    propTypes: {
+        images: React.PropTypes.array.isRequired,
+        selectedImages: React.PropTypes.array.isRequired,
+        columns: React.PropTypes.number,
+        padding: React.PropTypes.number,
+        onSelect: React.PropTypes.func,
+        onDeselect: React.PropTypes.func
+    },
```
Comment (hcchien) [inline admin/client/components/ImagesSelection.js]: key sorting

File: fields/types/imagerelationship/ImageRelationshipField.js [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+    },
+
+    loadById (id) {
+        const _this = this;
+        return new Promise(function(resolve, reject) {
+            if (_this._itemsCache[id]) {
+                return resolve(_this._itemsCache[id]);
+            }
+            xhr({
+                url: Keystone.adminPath + '/api/' + _this.props.refList.path + '/' + id,
```
Comment (hcchien) [inline fields/types/imagerelationship/ImageRelationshipField.js]: Do you want to have another variable here to avoid hard code the 'api' here?

**Comment by nickhsine** [comment]
@hcchien  if the code is ok, I need to merge those commits into one single commit.

**Comment by hcchien** [comment]
ok, merge it.

**Comment by nickhsine** [comment]
@hcchien Those commits are merged.
We can merge the PR if code review is ok.

**Comment by hcchien** [comment]
ok

**Pull request merged**